MobilePhone/src/yo/m/com/ysupov/entity/Shelf.java

```diff
@@ -0,0 +1,15 @@
+package yo.m.com.ysupov.entity;
+
+public abstract class Shelf implements Furniture, Size {
```

Comment:
As any furniture has dimensions I would move _implements Size_ to _Furniture_ interface